backend/src/user/user.service.ts

```diff
@@ -1,43 +1,26 @@
-import { Injectable, HttpException } from '@nestjs/common';
+import { Injectable, HttpException, ForbiddenException } from '@nestjs/common';
+import { User } from '@prisma/client';
 import { PrismaService } from 'nestjs-prisma';
+import { UpdateDto } from 'src/auth/dto';
+import * as argon from 'argon2';
 
 @Injectable()
 export class UserService {
   constructor(private readonly prisma: PrismaService) {}
 
-  //   async createFourtyTwoUser(
-  //     email: string,
-  //     userName: string,
-  //     fourtyTwoLogin: string,
-  //     password: string,
-  //   ): Promise<string> {
-  //     const response = await this.prisma.user.create({
-  //       data: {
-  //         email: email,
-  //         userName: fourtyTwoLogin,
-  //         password: password,
-  //         isFourtyTwoStudent: true,
-  //       },
-  //     });
-  //     return '42 user created!\n';
-  //   }
+  async updateUser(user: User, dto: Partial<UpdateDto>) {
+    if (dto.hash){
+      if (user.isFourtyTwoStudent)
+        throw new ForbiddenException("Can't change 42 password");
+      const hash = await argon.hash(dto.hash);
+      dto.hash = hash;
+    }
```

Comment:
User can update all the fields. Might need to filter the "body" data sent by the user.